sihl/test/web.ml

```diff
@@ -1,6 +1,62 @@
 open Lwt.Syntax
 open Alcotest_lwt
 
+let combine_two_routers_builds_paths _ () =
+  let router =
+    Sihl.Web.(
+      combine
+        "/root"
```

Comment:
Let's think if we want users to have to explicitly add slashes to their routes. They might make unintended routes this way if they for example forget a slash infront of a subroute. It's also not clear what happens if you forget the slash on the root route.

Reply:
How about something like this:
```ocaml
let handler _ = Lwt.return @@ Sihl.Web.Response.of_plain_text ""

let router =
  Sihl.Web.(
    choose
      [ choose
          ~path:"orders"
          [ get "" handler; get ":id" handler; delete ":id" handler ]
      ; choose
          "users"
          [ get "" handler
          ; choose
              ":id"
              [ get "" handler; post ":foo" handler; get "orders" handler ]
          ]
      ])
;;
```

Reply:
It's a little verbose, what do you think about removing the `~path` argument and making `choose` an infix operator? 
```ocaml
let router =
  Sihl.Web.(
    "/"
      >~>
      [ "orders"
          >~> [ get "" handler; get ":id" handler; delete ":id" handler ]
      ; "users"
          >~>
          [ get "" handler
          ;
              ":id"
                >~>
              [ get "" handler; post ":foo" handler; get "orders" handler ]
          ]
      ])
;;
```

Reply:
I adjusted the API now